Todo.Infrastructure/Services/TodoService.cs

```diff
@@ -0,0 +1,144 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using Todo.Application.Interfaces;
+using Todo.Domain.Entities;
+using Todo.Domain.Helpers;
+using Todo.Domain.Mappers;
+using Todo.Domain.Models;
+
+namespace Todo.Infrastructure.Services
+{
+    public class TodoService : ITodoService
+    {
+        private readonly AppDbContext _context;
+        private readonly IHttpContextAccessor _httpContext;
+        private readonly IUserResolverService _userResolverService;
+        public TodoService(AppDbContext context, IHttpContextAccessor httpContext, IUserResolverService userResolverService)
+        {
+            _context = context;
+            _httpContext = httpContext;
+            _userResolverService = userResolverService;
+        }
+        public async Task<TodoList> AddItem(ToDoModel item)
+        {
+            try
+            {
+                var userIdClaim = _httpContext.HttpContext!.User.FindFirstValue("id");
+                bool parseSuccess = Guid.TryParse(userIdClaim, out Guid userId);
+                if (!parseSuccess) throw new ApplicationException("Invalid user id");
+
+                var toDoItem = TodoMapper.AddTodoItem(item, userId);
+                _context.TodoLists.Add(toDoItem);
+                await _context.SaveChangesAsync();
+                return toDoItem;
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+        }
+
+        public async Task<TodoList?> GetItemById(Guid id)
+        {
+            try
+            {
+                var role = _userResolverService.GetUserRole();
+                var userId = _userResolverService.GetUserId();
+                TodoList? item;
+                if (string.Equals(role, UserRoleType.User.ToString()))
+                {
+                    item = await _context.TodoLists.FirstOrDefaultAsync(x => x.Id == id && x.UserId == userId);
```

Comment:
Please design and use some Middleware to intercept the Call to particular EndPoint. This Middle ware shall be responsible to allow / restrict user from accessing certain endpoints on the basis of resource ownership

Reply:
Done. Added middleware for checking user access

Reply:
I have removed the middleware and added an Action Filter for checking user access